src/sqlfluff/dialects/dialect_soql.py

```diff
@@ -41,11 +39,53 @@
     "NEXT_FISCAL_YEAR",
 ]
 
-soql_dialect.sets("reserved_keywords").update(date_literals)
+date_n_literals = [
+    "LAST_N_DAYS",
+    "NEXT_N_DAYS",
+    "LAST_N_WEEKS",
+    "NEXT_N_WEEKS",
+    "LAST_N_MONTHS",
+    "NEXT_N_MONTHS",
+    "LAST_N_QUARTERS",
+    "NEXT_N_QUARTERS",
+    "LAST_N_YEARS",
+    "NEXT_N_YEARS",
+    "LAST_N_FISCAL_QUARTERS",
+    "NEXT_N_FISCAL_QUARTERS",
+    "LAST_N_FISCAL_YEARS",
+    "NEXT_N_FISCAL_YEARS",
+]
+
+soql_dialect.sets("reserved_keywords").update(date_literals + date_n_literals)
```

Comment:
Are these definitely reserved keywords? Or they can be used as column names in other dialects so not reserved,  but not sure if that's true here.

Reply:
I would treat as reserved for now (though I'm fine to swap it if you prefer). I am not seeing documentation either way and don't have access to a Salesforce instance to test it.